cobigen/cobigen-openapiplugin-parent/cobigen-openapiplugin/src/main/java/com/devonfw/cobigen/openapiplugin/inputreader/OpenAPIInputReader.java

```diff
@@ -413,33 +415,53 @@ private PropertyDef extractReferenceProperty(OpenApi3 openApi, Schema componentS
      * @return list of {@link PathDef}'s
      */
     private List<PathDef> extractPaths(Map<String, ? extends Path> paths, String componentName) {
+        Matcher matcher;
+        Pattern pattern;
+        String match, root_component = null, version = null;
+
         List<PathDef> pathDefs = new LinkedList<>();
+
         for (String pathKey : paths.keySet()) {
             if (pathKey.toLowerCase().contains(componentName.toLowerCase())) {
                 String[] mp = pathKey.split("/");
                 String pathUri = "/";
-                for (int i = 3; i < mp.length; i++) {
-                    pathUri = pathUri.concat(mp[i] + "/");
+
+                try {
+                    match = "^\\/[^\\/]+\\/+[^\\/]+\\/(.+)";
+                    pattern = Pattern.compile(match);
+                    matcher = pattern.matcher(pathKey);
+                    matcher.find();
+                    pathUri += matcher.group(1);
+                    if (pathUri.charAt(pathUri.length() - 1) != "/".charAt(0)) {
+                        pathUri += "/";
+                    }
+                } catch (IllegalStateException e) {
+                    ;
                 }
-                PathDef path = new PathDef(pathUri, mp[2]);
-                if (pathKey.toLowerCase().contains(mp[1].toLowerCase())) {
-                    for (String opKey : paths.get(pathKey).getOperations().keySet()) {
-                        OperationDef operation = new OperationDef(opKey);
-                        operation.setDescription(paths.get(pathKey).getOperation(opKey).getDescription());
-                        operation.setSummary(paths.get(pathKey).getOperation(opKey).getSummary());
-                        operation.setOperationId((paths.get(pathKey).getOperation(opKey).getOperationId()));
-                        operation.setResponses(extractResponses(paths.get(pathKey).getOperation(opKey).getResponses(),
-                            paths.get(pathKey).getOperation(opKey).getTags()));
-                        operation.setTags(paths.get(pathKey).getOperation(opKey).getTags());
-                        if (path.getOperations() == null) {
-                            path.setOperations(new ArrayList<OperationDef>());
-                        }
-                        operation.getParameters()
-                            .addAll(extractParameters(paths.get(pathKey).getOperation(opKey).getParameters(),
-                                paths.get(pathKey).getOperation(opKey).getTags(),
-                                paths.get(pathKey).getOperation(opKey).getRequestBody()));
-                        path.getOperations().add(operation);
+
+                root_component = mp[1];
```

Comment:
`root_component` is not being used anywhere. It would be very interesting to add a new parameter on `PathDef`, like the following:

`PathDef path = new PathDef(rootComponent, pathUri, version);`

This would fix this issue #878 

Also remember that Java is Camel Case ;), it should be `rootComponent`.